Library/Homebrew/command.rb

```diff
@@ -0,0 +1,157 @@
+module Homebrew
+  class Command
+    attr_reader :command_name, :valid_options, :description, :help_output,
+      :man_output
+
+    def initialize_variables
+      @valid_options = []
+      @root_options = []
+      @argv = ARGV.dup
+    end
+
+    def options(&block)
+      initialize_variables
+      @parent = nil
+      instance_eval(&block)
+      generate_help_and_manpage_output
+    end
+
+    def cmd_name(cmd)
+      @command_name = cmd
+    end
+
+    def add_valid_option(option_hash)
+      @valid_options.push(option_hash)
+    end
+
+    def option(*args, **option_hash, &block)
+      option_hash[:option] = "--#{args[0]}"
```

Comment:
@MikeMcQuaid, this function basically stores each option provided in the DSL in an array `@valid_options` by handling the parent-child relationship of options as provided in the nested (or un-nested) `option` block in the DSL.
Details:
So we have an array `@valid_options` that stores each option as a hash. If the `input option` provided in the DSL has no parent option, then this function just stores that `input option` in the array `@valid_options`. If, however, `input option` has a parent option, then this function updates that parent option to indicate that `input option` is it's child.